engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -315,9 +316,22 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId,
     case Event(FailedNonRetryableResponse(jobKey, reason, returnCode), stateData) =>
       pushFailedJobMetadata(jobKey, returnCode, reason, retryableFailure = false)
       saveUnsuccessfulJobResults(jobKey, returnCode, reason)
-      context.parent ! WorkflowExecutionFailedResponse(stateData.executionStore, stateData.outputStore, List(reason))
       val mergedStateData = stateData.mergeExecutionDiff(WorkflowExecutionDiff(Map(jobKey -> ExecutionStatus.Failed)))
-      goto(WorkflowExecutionFailedState) using mergedStateData.removeBackendJobExecutionActor(jobKey)
+                                     .removeBackendJobExecutionActor(jobKey)
+      if (workflowDescriptor.getWorkflowOption(WorkflowFailureMode).contains(ContinueWhilePossible.toString)) {
+        mergedStateData.workflowCompletionStatus match {
+          case Some(completionStatus) if completionStatus == Failed =>
+            goto(WorkflowExecutionFailedState) using mergedStateData
+          case Some(completionStatus) if completionStatus == Done =>
+            workflowLogger.info(mergedStateData.outputsJson())
+            goto(WorkflowExecutionSuccessfulState) using mergedStateData
```

Comment:
Discussed in person but just not to forget about it: It seems strange to have a `WorkflowExecutionSuccessfulState` here since we're in the `FailedNonRetryableResponse` case.